src/main/java/edu/byu/cs/util/RepoUrlValidator.java

```diff
@@ -1,20 +1,27 @@
 package edu.byu.cs.util;
 
 import edu.byu.cs.autograder.GradingException;
+import edu.byu.cs.autograder.git.GitHelper;
 import org.eclipse.jgit.annotations.Nullable;
 
+import java.io.File;
+import java.util.UUID;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class RepoUrlValidator {
 
     public static boolean isValid(@Nullable String repoUrl) {
+        File cloningDir = new File("./tmp" + UUID.randomUUID());
         try {
-            clean(repoUrl);
+            GitHelper.fetchRepoFromUrl(repoUrl, cloningDir);
             return true;
-        } catch (InvalidRepoUrlException e) {
+        } catch (GradingException e) {
```

Comment:
I would recommend moving this closing logic into a separate function with the call signature: `static boolean canClone(String repoUrl)`. [Notice it doesn't throw an error.] It would be good to do a similar transformation to create a `static boolean canClean()` method.

Then the body of the `isValid()` method can be streamlined and improved in readability to `return canClean() && canClone()`.

Reply:
So then it'd be instead of catching exceptions, these two methods catch them and return Boolean, and then I return false if either of them is false?

Reply:
Also, written here or in githelper?

Reply:
Yes, you got it.

The two functions `canClean()` and `canClone()` effectively call other functions to do the work, but then interpret the result to always return a `boolean` value. In this case, they will be similar since they are both catching a specific kind of error to interpret as `false`; however, they are different because they are catching different errors. 

Additional checks in the future can also be introduced to return a `boolean` value, but they may not be catching errors (they may just be evaluating conditions). With the `isFork()` function that will be coming soon, it will make a network call, and then just report the result. Compare https://github.com/softwareconstruction240/softwareconstruction/issues/138.

This family of validation functions (`canClean()`, `canClone()`, and potentially `isFork()`)  belongs best in the `RepoUrlValidator` class. Here are a few reasons:
* They are each adapting logic from other places into a format that works well for the the `RepoUrlValidator.isValid()` function. Therefore, they belong with that function
* Notice that in most cases, the actual "work" does live in other classes and is being delegated to. That's a good thing. For example, the `canClean()` calls an instance method of the class, `canClone()` calls a method on a different class, and `isFork()` will ask a remote server for information and extract just the flag it cares about.
* Modifying the source locations for each of these isn't appropriate, or possible in all cases.